samples/TerraFX/Graphics/HelloTextureTransform.cs

```diff
@@ -0,0 +1,204 @@
+            using (var textureStagingBuffer = graphicsDevice.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Default, GraphicsResourceCpuAccess.Write, 64 * 1024 * 4))
+            {
+                var currentGraphicsContext = graphicsDevice.CurrentContext;
+
+                currentGraphicsContext.BeginFrame();
+                _trianglePrimitive = CreateTrianglePrimitive(currentGraphicsContext, vertexStagingBuffer, textureStagingBuffer);
+                currentGraphicsContext.EndFrame();
+
+                graphicsDevice.Signal(currentGraphicsContext.Fence);
+                graphicsDevice.WaitForIdle();
+            }
+        }
+
+        protected override unsafe void Update(TimeSpan delta)
+        {
+            const float translationSpeed = MathF.PI;
+
+            float radians = _trianglePrimitiveTranslationAngle;
+            {
+                radians += (float)(translationSpeed * delta.TotalSeconds);
+                radians = radians % (2 * MathF.PI); 
+            }
+            _trianglePrimitiveTranslationAngle = radians;
+            float x = 0.5f * MathF.Cos(radians);
+            float y = 0.5f * MathF.Sin(radians);
+
+
+            var constantBuffer = (GraphicsBuffer)_trianglePrimitive.InputResources[0];
+            var pConstantBuffer = constantBuffer.Map<Matrix4x4>();
+
+            // Shaders take transposed matrices, so we want to set X.W
+            pConstantBuffer[0] = new Matrix4x4(
+                new Vector4(1.0f, 0.0f, 0.0f, x),
+                new Vector4(0.0f, 1.0f, 0.0f, y),
+                new Vector4(0.0f, 0.0f, 1.0f, 0),
+                new Vector4(0.0f, 0.0f, 0.0f, 1.0f)
+            );
+
+            constantBuffer.Unmap(0..sizeof(Matrix4x4));
+        }
+
+        protected override void Draw(GraphicsContext graphicsContext)
+        {
+            graphicsContext.Draw(_trianglePrimitive);
+            base.Draw(graphicsContext);
+        }
+
+        private unsafe GraphicsPrimitive CreateTrianglePrimitive(GraphicsContext graphicsContext, GraphicsBuffer vertexStagingBuffer, GraphicsBuffer textureStagingBuffer)
+        {
+            var graphicsDevice = GraphicsDevice;
+            var graphicsSurface = graphicsDevice.Surface;
+
+            var graphicsPipeline = CreateGraphicsPipeline(graphicsDevice, "TextureTransform", "main", "main");
+            var vertexBuffer = CreateVertexBuffer(graphicsContext, vertexStagingBuffer, aspectRatio: graphicsSurface.Width / graphicsSurface.Height);
+
+            var inputResources = new GraphicsResource[3] {
+                CreateConstantBuffer(graphicsContext),
+                CreateConstantBuffer(graphicsContext),
+                CreateTexture2D(graphicsContext, textureStagingBuffer),
+            };
+            return graphicsDevice.CreatePrimitive(graphicsPipeline, new GraphicsBufferView(vertexBuffer, vertexBuffer.Size, SizeOf<IdentityVertex>()), indexBufferView: default, inputResources);
```

Comment:
```suggestion
            return graphicsDevice.CreatePrimitive(graphicsPipeline, new GraphicsBufferView(vertexBuffer, vertexBuffer.Size, SizeOf<TextureVertex>()), indexBufferView: default, inputResources);
```

Same with other instances of `IdentityVertex` in the file.

Reply:
done